### PR TITLE
examples: add echo proto

### PR DESCRIPTION
**Pull request by menghanl**

### CODE REVIEW

File: examples/features/proto/echo/echo.proto [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+message EchoResponse {
+  string message = 1;
+}
+
+service Echo {
+  rpc UnaryEcho(EchoRequest) returns (EchoResponse) {}
+  rpc ServerStreamingEcho(EchoRequest) returns (stream EchoResponse) {}
+  rpc ClientStreamingEcho(stream EchoRequest) returns (EchoResponse) {}
+  rpc BidirectionalStreamingEcho(stream EchoRequest) returns (stream EchoResponse) {}
+}
```
Comment (dfawley) [inline examples/features/proto/echo/echo.proto]: Add this newline?

File: examples/features/proto/doc.go [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package proto
```
Comment (dfawley) [inline examples/features/proto/doc.go]: Needs copyright.

File: examples/features/proto/echo/echo.proto [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+syntax = "proto3";
+
+package echo;
```
Comment (dfawley) [inline examples/features/proto/echo/echo.proto]: `grpc.examples.features.proto.echo` ?
`grpc.examples.echo` ?

Comment (menghanl) [reply examples/features/proto/echo/echo.proto]: `grpc.examples.echo` should be good.

I didn't add `option go_package = `, I think we don't need it for this.

Comment (menghanl) [reply examples/features/proto/echo/echo.proto]: And the directory name needs to be renamed to `grpc_examples_echo`, to match the go package name.

**Pull request merged**